UT4MasterServer/Other/EpicID.cs

```diff
@@ -28,11 +27,30 @@ private EpicID(string id)
 
 		ID = id.ToLower();
 
-		foreach (char c in ID)
+		if (!IsHexString(ID))
+		{
+			throw new ArgumentException("id needs to be a hexadecimal string");
+		}
+	}
+
+	public static bool IsHexString(string input)
+	{
+		if (string.IsNullOrWhiteSpace(input))
+		{
+			return false;
+		}
+
+		input = input.ToLower();
```

Comment:
Now `string.ToLower` is called twice for no reason when `EpicID.IsHexString` is called from constructor.
I think there should be an additional condition for uppercase characters in the foreach loop below instead of converting whole string to lower *just* to have one condition less.

Reply:
Done